core/src/main/java/org/geysermc/geyser/network/netty/Bootstraps.java

```diff
@@ -81,7 +94,7 @@ public static void setupBootstrap(AbstractBootstrap bootstrap) {
         }
     }
 
-    private static int[] fromString(String ver) {
+    private static int[] fromString(String ver) throws IllegalArgumentException {
         String[] parts = ver.split("\\.");
```

Comment:
Imo seems like the issue is here... could we not just easily use a regex along the lines of:

```java
    public static int[] fromString(String input) {
        // Match only beginning of string for at least two, up to three digits separated by dot 
        Pattern pattern = Pattern.compile("^(\\d+)\\.(\\d+)(?:\\.(\\d+))?");
        Matcher matcher = pattern.matcher(input);
        
        int[] version = {0, 0, 0};
        
        if (matcher.find()) {
            version[0] = Integer.parseInt(matcher.group(1));
            version[1] = Integer.parseInt(matcher.group(2));
            
            if (matcher.group(3) != null) {
                version[2] = Integer.parseInt(matcher.group(3));
            }
        }
        
        return versionNumbers;
    }
```

This avoids having to throw exceptions as the default will simply be 0, which will not meet the criteria.

Reply:
Compile the pattern at class level if we are calling this multiple times

Reply:
sure, that also works - I'll adjust it once I'm home later. Feel free to push to this branch if that's more convenient though

Reply:
> Compile the pattern at class level if we are calling this multiple times

This would only ever be called once during the setup of the Bootstrap class